@navikt/core/react/src/layout/box/Box.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { forwardRef } from "react";
+import { OverridableComponent } from "../../util/OverridableComponent";
+import cl from "clsx";
+import {
+  ResponsiveProp,
+  SpacingScale,
+  getResponsivePropsPaddingForInlineOrBlock,
+} from "../utilities/css";
+import { BackgroundColors, BorderColors, BorderRadii, Shadows } from "./types";
+
+type Element = "div" | "span" | "section" | "legend" | "ul" | "li";
+
+type Spacing = ResponsiveProp<SpacingScale>;
+
+export interface BoxProps extends React.HTMLAttributes<HTMLDivElement> {
+  children?: React.ReactNode;
+  /** HTML Element type
+   * @default 'div'
+   */
+  as?: Element;
+  /** Background color */
+  background?: BackgroundColors;
+  /** Background color on hover */
+  backgroundHover?: BackgroundColors;
+  /** Border color */
+  borderColor?: BorderColors;
+  /** Border color on hover */
+  borderColorHover?: BorderColors;
+  /** Border radius */
+  borderRadius?: BorderRadii;
+  /** Vertical end horizontal start border radius */
+  borderRadiusEndStart?: string;
+  /** Vertical end horizontal end border radius */
+  borderRadiusEndEnd?: string; //TODO
+  /** Vertical start horizontal start border radius */
+  borderRadiusStartStart?: string; //TODO
+  /** Vertical start horizontal end border radius */
+  borderRadiusStartEnd?: string; //TODO
+  /** Spacing around children. Accepts a spacing token or an object of spacing tokens for different screen sizes.
+   * @example
+   * padding='4'
+   * padding={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  padding?: Spacing;
+  // TODO
+  paddingInline?: Spacing;
+  paddingBlock?: Spacing;
+  /** Vertical start spacing around children. Accepts a spacing token or an object of spacing tokens for different screen sizes.
+   * @example
+   * paddingBlockStart='4'
+   * paddingBlockStart={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingBlockStart?: Spacing;
+  /** Vertical end spacing around children. Accepts a spacing token or an object of spacing tokens for different screen sizes.
+   * @example
+   * paddingBlockEnd='4'
+   * paddingBlockEnd={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingBlockEnd?: Spacing;
+  /** Horizontal start spacing around children. Accepts a spacing token or an object of spacing tokens for different screen sizes.
+   * @example
+   * paddingInlineStart='4'
+   * paddingInlineStart={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingInlineStart?: Spacing;
+  /** Horizontal end spacing around children. Accepts a spacing token or an object of spacing tokens for different screen sizes.
+   * @example
+   * paddingInlineEnd='4'
+   * paddingInlineEnd={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingInlineEnd?: Spacing;
+  /** Shadow on box */
+  shadow?: Shadows;
```

Comment:
Kunne dette API-et brukes her? Slipper da prop-en `shadowHover` 🤔 
```suggestion
  shadow?: Shadows | {default: Shadows, hover: Shadows};
```
Kanskje også `{default: Shadows, hover: Shadows, focus: Shadows}` da, men 🤷

Reply:
Da skriver jeg om likt på alle `:hover` props da :)

Reply:
Får også nå lyst til å collapse props for borderRadius og padding 🤔 ... men hva burde et slikt object se ut som? (`default` gir ikke like mye mening her som ved hover)

example:
```js
  /** Border radius */
  borderRadius?:
    | BorderRadii
    | {
        default?: BorderRadii;
        left: BorderRadii;
        right: BorderRadii;
        top: BorderRadii;
        bottom: BorderRadii;
      };
```

Reply:
og tenker også at jeg burde lage en translation for `xl` -> `xlarge`, `sm` -> `small` osv 🤔  (før vi lanserer APIet / komponenten)